Character.h

```diff
@@ -5,18 +5,34 @@
 #include <fstream>
 
 class Character {
-private:
+protected:
 	const std::string name;
 	int hp;
-	const int dmg;
+	int dmg;
 public:
-	Character(const std::string name, const int hp, const int dmg);
+	Character(const std::string name, const int hp, int dmg);
+	virtual ~Character();
 	std::string getName() const;
 	int getHp() const;
 	int getDmg() const;
-	void DMGTaken(const Character character);
 	bool IsDead() const;
-	static Character parseUnit(const std::string& FileName);
+	void DMGTaken(Character*);
+	virtual void OnePunch(Character* foe);
+	static Character* parseUnit(const std::string& FileName);
+	void Fight(Character*);
```

Comment:
Inkább referenciaként legyenek átadva a Character paraméterek, és ahol lehet, ott const refként.

Reply:
@oliverosz Csupán ez az egy javítás van még hátra azokból amit megírtál, azonban kérdezni szeretnék ezzel kapcsolatban: hogy a mutatókat cseréljem ki referenciákra, vagy maradjanak a mutatók, csak referenciát is kapjanak?
Köszönöm előre is a válaszod!

Reply:
@Szabi1104 Úgy érted, referenciaként adnád át a mutatót? Annak itt nem sok értelme lenne. A mutató helyett legyen referencia, mert a mutató lehet nullptr is, ami viszont nincs ellenőrizve. Ha referenciát vár, akkor muszáj egy létező objektumot megadni neki.